tests/integration/test_dynamic_sampling.py

```diff
@@ -342,124 +342,6 @@ def test_it_keeps_events(mini_sentry, relay, rule_type, event_factory):
         mini_sentry.captured_outcomes.get(timeout=2)
 
 
-@pytest.mark.parametrize("should_remove", [True, False])
-@pytest.mark.parametrize(
-    "rule_type, event_factory",
-    [
-        ("error", _create_event_envelope),
-        ("transaction", _create_transaction_envelope),
-        ("trace", _create_transaction_envelope),
-    ],
-)
-def test_bad_dynamic_rules_in_processing_relays(
-    mini_sentry,
-    relay_with_processing,
-    events_consumer,
-    transactions_consumer,
-    should_remove,
-    rule_type,
-    event_factory,
-):
-    """
-    Configurations that contain bad (unrecognized) rules should be handled by
-    removing the offending rules and sampling using the correct rules
-    """
-
-    sample_rate = 0 if should_remove else 1
-
-    relay = relay_with_processing()
-    project_id = 42
-    config = mini_sentry.add_full_project_config(project_id)
-    public_key = config["publicKeys"][0]["publicKey"]
-
-    if rule_type == "error":
-        consumer = events_consumer()
-    else:
-        consumer = transactions_consumer()
-
-    # add a bad condition (with the opposite sample rate to make it evident it is not applied)
-    rules = _add_sampling_config(
-        config, sample_rate=1 - sample_rate, rule_type=rule_type
-    )
-    last_rule = rules[-1]
-    last_rule["condition"]["inner"].append(
-        {
-            "op": "BadOperator",
```

Comment:
One thing we did not verify in the Rust test is whether `BadOperator` is indeed parsed into `Unsupported`. So we should add a rust unit test that tests only the parsing, something along the lines of https://github.com/getsentry/relay/blob/60d4305600372407d619a239f42ec8c295bb2f5e/relay-metrics/src/aggregation.rs#L2170-L2207.